### PR TITLE
feat: adds cancel retirement feature

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2521](https://openedx.atlassian.net/browse/PROD-2521)
Adds a new feature to cancel a retirement of a user account that is under 14 days cooloff period. 

Screenshot 2022-08-26 at 2 58 10 AM

Screenshot 2022-08-26 at 2 58 18 AM

### CODE REVIEW

File: src/users/data/api.js [hunk 1]
```diff
@@ -351,11 +351,20 @@ export async function getOnboardingStatus(username) {
...
   let user = null;
+  let retirementStatus = null;
+  let errorResponse = null;
   try {
     user = await getUser(userIdentifier);
   } catch (error) {
     if (error.userError) {
       errors.push(error.userError);
+      errorResponse = JSON.parse(error.customAttributes.httpErrorResponseData);
+      if (errorResponse?.can_cancel_retirement) {
```
Comment (DawoudSheraz) [inline src/users/data/api.js:362]: `?.`? What's this notation?

File: src/users/data/api.js [hunk 2]
```diff
@@ -351,11 +351,20 @@ export async function getOnboardingStatus(username) {
...
+        retirementStatus = {
```
Comment (DawoudSheraz) [inline src/users/data/api.js:363]: Unit test coverage missing.

File: src/users/UserPage.jsx [hunk 3]
```diff
@@ -159,6 +163,12 @@ export default function UserPage({ location }) {
           changeHandler={handleUserSummaryChange}
         />
       )}
+      {!loading && data.canCancelRetirement && (
+        <CancelRetirement
+          retirementId={data.retirementId}
+          changeHandler={handleUserSummaryChange}
+        />
+      )}
```
Comment (DawoudSheraz) [inline src/users/UserPage.jsx:171]: missing unit test coverage.

**Review by DawoudSheraz — COMMENTED** [review]
- To add missing unit tests for API and UserPage

File: src/users/data/api.js (hunk 1 above)
Comment (Ali-D-Akbar) [reply src/users/data/api.js:362]: if (errorResponse !== null && errorResponse. can_cancel_retirement)

File: src/users/UserPage.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/UserPage.jsx:171]: There's no unit test made for UserPage. Are we planning to add these tests someday?
Also for some reasons, I'm not seeing #L167 left out from the coverage.

Screenshot 2022-09-07 at 3 15 27 AM

**Review by DawoudSheraz — APPROVED** [review]
I didn't test the changes locally. Overall, looks good.

**Review by AliAdnanSohail — APPROVED** [review]

**Pull request merged**